src/Npgsql/PostgresException.cs

```diff
@@ -323,7 +325,7 @@ public override IDictionary Data
                     where v != null
                     where k != nameof(Position) && k != nameof(InternalPosition) || (int)v != 0
                     select new { Key = k, Value = v }
-                    ).ToDictionary(kv => kv.Key, kv => kv.Value)
+                    ).ToDictionary(kv => (object)kv.Key, kv => kv.Value)
```

Comment:
Please remove casting to an `object` and add generic parameters to the method, i.e. `ToDictionary<object, object>()`. It will be more obvious.

Reply:
I agree on the suggestion. However, the overload `.ToDictionary<TElement, TKey, TValue>` has three generic parameters where `TElement` is the anonymous type implicitely created at the preceding line and I cannot pass that explicitely.

I could instead yield `KeyValuePair<object, object>` in the Linq expression. Would you prefer that for readability?

Reply:
Ah so... No leave it as is (: